### PR TITLE
Duplicate connectors on ctrl + click on input ports

### PR DESCRIPTION
**Pull request by yeexinc**
### Purpose

Duplicate connectors when the user does a ctrl + click on input ports.
duplicatectrl

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@Benglin 
@riteshchandawar

### CODE REVIEW

**Comment by riteshchandawar** [comment]
Functionality wise this is what I was looking, so LGTM from my side. 

@Benglin Can you please do the code review? 

Thanks,
Ritesh

File: src/DynamoCore/Models/DynamoModelCommands.cs [hunk 1]
```diff
@@ -271,6 +271,10 @@ void MakeConnectionImpl(MakeConnectionCommand command)
                     EndShiftReconnections(nodeId, command.PortIndex, command.Type);
                     break;
 
+                case MakeConnectionCommand.Mode.BeginCtrlConnection:
+                    BeginCtrlConnection(nodeId, command.PortIndex, command.Type);
+                    break;
+
                 case MakeConnectionCommand.Mode.EndAndStartCtrlConnection:
                     EndAndStartCtrlConnection(nodeId, command.PortIndex, command.Type);
                     break;
```
Comment (Benglin) [inline src/DynamoCore/Models/DynamoModelCommands.cs]: Hi @yeexinc, after further considerations I think better names for the modes would be `BeginDuplicateConnection` and `BeginCreateConnections`. Please also rename the methods to reflect this.

File: src/DynamoCore/Models/DynamoModelCommands.cs [hunk 2]
```diff
@@ -315,6 +319,32 @@ void BeginConnection(Guid nodeId, int portIndex, PortType portType)
...
+            {   
+                // If the port doesn't have any existing connections, begin connection as per normal
+                BeginConnection(nodeId, portIndex, portType);
+                return;
+            }
+            // Duplicate the existing connection
+            activeStartPorts = new PortModel[] { portModel.Connectors[0].Start };
+            firstStartPort = portModel.Connectors[0].Start;
+        }
+
```
Comment (Benglin) [inline src/DynamoCore/Models/DynamoModelCommands.cs]: Will the undo work for each of the connectors created?

Comment (yeexinc) [inline src/DynamoCore/Models/DynamoModelCommands.cs]: @Benglin - Yes, each undo action corresponds to one connector created.

**Comment by Racel** [comment]
@QilongTang - I think we just need to add this to our backlog. Let us just try to merge this in if you think it looks good to go. Adding a task for this.

**Comment by alfarok** [comment]
closing and picking this work up in #9567

**Pull request closed without merging**